src/NHibernate/Type/AbstractStringType.cs

```diff
@@ -57,14 +57,15 @@ public AbstractStringType(SqlType sqlType)
 		public override void Set(DbCommand cmd, object value, int index, ISessionImplementor session)
 		{
 			var parameter = cmd.Parameters[index];
+			var stringValue  = Convert.ToString(value);
```

Comment:
> I fear that the general approach using Convert.ToString might cover programming bugs of app developers. Convert.ToString(new object()) leads to "System.Object", the result of Convert.ToString(1.5m) is locale-dependent.

I totally agree with @csharper2010 on this. We should not silently convert parameters with incorrect types to string. I see it as a source of hard to find bugs